### DISCUSSION TITLE
SAML Authentication

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1970] [Q&A]
Hi,

SAML Authentication is available in Oqtane?

1. User passwords should not be saved in Oqtane database
2. Validate user credentials against AD or external database (NOT Oqtane DB) while on user login
3. User details (user Full Name, designation, etc.,) can be sync using background / scheduled jobs

How it can be possible and what are the ways to achieve in Oqtane?

Thanks & Regards!

**Comment by sbwalker** [comment]
@Sharafudeen SAML support is currently not available in Oqtane - enhancements to authentication are on the roadmap for 3.1 ( see https://github.com/oqtane/oqtane.framework#roadmap ).

1. Currently Oqtane uses .NET Core Identity - the standard/approved library from Microsoft for handling local user accounts in applications. In this scenario authentication is handled locally and passwords are NOT saved in the database - only their hashed form are stored.

2/3. Are intended to be included in 3.1 release

**Reply by Sharafudeen** [reply]
How can I validate a user against AD once success then validate with only username in Oqtane DB (without password hash)?

I don't want to keep user credentials in Oqtane DB this is because of some internal policy which is user credentials only available in AD.

**Reply by sbwalker** [reply]
As I mentioned above, Oqtane currently does not have support for this scenario. You would need to figure out the core customizations yourseif. I would encourage you to read https://www.oqtane.org/blog/!/14/exploring-authentication-in-blazor as auth is complicated in Blazor.

**Reply by Sharafudeen** [reply]
Oqtane 3.1 version having AD authentication instead of validating against Oqtane database? If so, how can I use it?

**Reply by sbwalker** [reply]
Oqtane 3.1 supports OpenId Connect and OAuth 2.0. This blog post explains the options - https://www.oqtane.org/blog/!/37/oauth-2-0-and-openid-connect
